pdr_backend/lake/table_bronze_pdr_slots.py

```diff
@@ -54,35 +54,50 @@ def get_bronze_pdr_slots_data_with_SQL(
+                {pdr_slots_table_name}.timestamp AS timestamp,
                 GREATEST(
                     {pdr_slots_table_name}.timestamp,
-                    {pdr_payouts_table_name}.timestamp,
-                    {etl_bronze_pdr_predictions_table_name}.timestamp,
-                    {pdr_truevals_table_name}.timestamp
-                ) as last_event_timestamp,
+                    {pdr_truevals_table_name}.timestamp,
+                    joined_{pdr_payouts_table_name}.timestamp,
+                    joined_{etl_bronze_pdr_predictions_table_name}.timestamp
+                ) AS last_event_timestamp
             FROM
                 {pdr_slots_table_name}
-            LEFT JOIN {pdr_payouts_table_name}
-            ON {pdr_slots_table_name}.ID = SPLIT_PART({pdr_payouts_table_name}.ID, '-', 1)
-                || '-' || SPLIT_PART({pdr_payouts_table_name}.ID, '-', 2)
-            LEFT JOIN {pdr_truevals_table_name}
-            ON {pdr_slots_table_name}.ID = {pdr_truevals_table_name}.ID
-            LEFT JOIN {etl_bronze_pdr_predictions_table_name}
-            ON {pdr_slots_table_name}.ID = {etl_bronze_pdr_predictions_table_name}.slot_id
+            LEFT JOIN {pdr_truevals_table_name} ON {pdr_slots_table_name}.ID = {pdr_truevals_table_name}.ID
+            LEFT JOIN (
+                SELECT 
+                    {pdr_slots_table_name}.ID AS slot_id,
+                    MAX({etl_bronze_pdr_predictions_table_name}.pair) AS pair,
+                    MAX({etl_bronze_pdr_predictions_table_name}.timeframe) AS timeframe,
+                    MAX({etl_bronze_pdr_predictions_table_name}.source) AS source,
+                    MAX({etl_bronze_pdr_predictions_table_name}.timestamp) AS timestamp
+                FROM {pdr_slots_table_name}
+                LEFT JOIN {etl_bronze_pdr_predictions_table_name} ON SPLIT_PART({pdr_slots_table_name}.ID, '-', 1) = {etl_bronze_pdr_predictions_table_name}.contract
+                GROUP BY {pdr_slots_table_name}.ID
+            ) AS joined_{etl_bronze_pdr_predictions_table_name} ON {pdr_slots_table_name}.ID = joined_{etl_bronze_pdr_predictions_table_name}.slot_id
+            LEFT JOIN (
+                SELECT 
+                    SPLIT_PART(ID, '-', 1) || '-' || SPLIT_PART(ID, '-', 2) AS slot_id,
+                    MAX(roundSumStakesUp) AS roundSumStakesUp,
+                    MAX(roundSumStakes) AS roundSumStakes,
+                    MAX(timestamp) AS timestamp
+                FROM {pdr_payouts_table_name}
+                GROUP BY slot_id
+            ) AS joined_{pdr_payouts_table_name} ON {pdr_slots_table_name}.ID = joined_{pdr_payouts_table_name}.slot_id
             WHERE
-                {pdr_slots_table_name}.timestamp >= {st_ms}
-                AND {pdr_slots_table_name}.timestamp <= {fin_ms}
+                pdr_slots.timestamp >= {st_ms}
+                AND pdr_slots.timestamp <= {fin_ms}
+            ORDER BY timestamp
         """
```

Comment:
1. please dump the data at the end o the DF
2. please copy this data into the test to make it easy to see
3. use the data from test to be processed by this function, such that we can:
a) see the output from this query
b) see the data being processed correctly